specs/experimental/alligator.md

```diff
@@ -0,0 +1,401 @@
+    - [`getPastTotalSupply`](#getpasttotalsupply)
+    - [`subdelegations`](#subdelegations)
+  - [Events](#events)
+    - [`Subdelegation`](#subdelegation)
+    - [`Subdelegations`](#subdelegations)
+    - [`DelegateVotesChanged`](#delegatevoteschanged)
+- [Storage](#storage)
+- [Types](#types)
+  - [`SubdelegationRule`](#subdelegationrule)
+  - [`AllowanceType`](#allowancetype)
+- [Backwards Compatibility](#backwards-compatibility)
+- [User Flow](#user-flow)
+  - [Partial delegations](#partial-delegations)
+  - [Constrained delegations](#constrained-delegations)
+  - [Redelegations](#redelegations)
+  - [Differences](#differences)
+- [Security Considerations](#security-considerations)
+  - [Dependence on Alligator](#dependence-on-alligator)
+  - [Connection with GovernanceToken](#connection-with-governancetoken)
+- [Future Considerations](#future-considerations)
+  - [Cross Chain Delegations](#cross-chain-delegations)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Overview
+
+| Constant | Value                                        |
+|----------|----------------------------------------------|
+| Address  | `0x4200000000000000000000000000000000000043` |
+
+The `Alligator` contract implements subdelegations for the [`GovernanceToken`](gov-token.md). Subdelegations enable
+advanced delegation use cases, such as partial, time-constrained & block-based delegations, and relative & fixed allowances.
+
+The `Alligator` contract migrates the delegation state from the `GovernanceToken` to itself
+through a hook-based approach. Specifically, the `GovernanceToken` calls the `Alligator` contract's
+`afterTokenTransfer` function after a token transfer. This enables the `Alligator` contract to consume the hook and update
+its delegation and checkpoint mappings accordingly. If either address involved in the transfer (`_from_` or `_to`) has not
+been migrated to the `Alligator` contract, the contract copies the address' checkpoint data from the
+`GovernanceToken` to its own state.
+
+## Interface
+
+### Core Functions
+
+#### `subdelegate`
+
+Allows subdelegation of token voting power to another address (delegatee) with a specified subdelegation rule. This function
+is intended to be called by users that require advanced delegation of the `GovernanceToken`.
+
+```solidity
+function subdelegate(address _delegatee, SubdelegationRule _rule) external
+```
+
+This function MUST check if the `msg.sender` or `_delegatee` addresses have been migrated by checking the `migrated` mapping
+from its [storage](#storage). If either address has not been migrated, the `Alligator` MUST copy the delegation
+and checkpoint data from the token contract to its own state. After copying the data, the `Alligator` MUST update
+the `migrated` mapping to reflect that the address has been migrated.
+
+Before updating the subdelegation, the `subdelegate` function MUST check the validity of the subdelegation rule. Specifically,
```

Comment:
Similarly to this suggestion, we could have a single paragraph that describes the following:

> This function MUST check if the `_delegator` or `_delegatee` addresses have been migrated by checking the `migrated` mapping
from its [storage](#storage). If either address has not been migrated, the `Alligator` MUST copy the delegation
and checkpoint data from the token contract to its own state. After copying the data, the `Alligator` MUST update
the `migrated` mapping to reflect that the address has been migrated.

Then link the paragraph in the multiple places where needed rather than copy/paste the text

Reply:
added the `Delegation Validation` and `Migration` sections.